xbmc/utils/EGLImage.cpp

```diff
@@ -183,7 +183,7 @@ bool CEGLImage::CreateImage(EglAttrs imageAttrs)
       }
       else
       {
-        if (eglAttrKey->first == EGL_LINUX_DRM_FOURCC_EXT)
+        if (eglAttrKey != eglAttributes.end() && eglAttrKey->first == EGL_LINUX_DRM_FOURCC_EXT)
```

Comment:
You are already in the "else" case of: "if (eglAttrValue != eglAttributes.end())" so that code now is actually deadcode by definition. The new condition check is always false.

Reply:
eglAttrValue or eglAttrKey?

Reply:
if (eglAttrKey != eglAttributes.end())
// do something
else
{
  if (**eglAttrKey != eglAttributes.end()** ...))

The bold case will never be true

Reply:
I am sorry - I take it back ... I parsed to fast: eglAttrValue vs. eglAttrKey